java/refactoring.java/src/org/netbeans/modules/refactoring/java/plugins/DeleteFile.java

```diff
@@ -75,11 +75,10 @@ public String getDisplayText() {
     public void performChange() {
         try {
             FileObject fo = URLMapper.findFileObject(res);
-            if (fo == null) {
-                throw new IOException(res.toString());
+            if (fo != null) {
+                id = BackupFacility.getDefault().backup(fo);
+                DataObject.find(fo).delete();
             }
-            id = BackupFacility.getDefault().backup(fo);
-            DataObject.find(fo).delete();
         } catch (DataObjectNotFoundException ex) {
```

Comment:
wondering why this change was necessary, this particular refactoring shouldn't delete anything, no?